src/amo/components/CollectionManager/index.js

```diff
@@ -0,0 +1,176 @@
+
+    if (!collection) {
+      // You'd have to click really fast to access a form without a
+      // collection so a user will not likely see this.
+      throw new Error(
+        'The form cannot be submitted without a collection');
+    }
+    if (!siteUserId) {
+      // The component won't be visible without a signed in user so a
+      // user will not likely see this.
+      throw new Error(
+        'The form cannot be submitted without a user');
+    }
+    if (!siteLang) {
+      // It is not possible to browse the site without a language so
+      // a user will not likely see this.
+      throw new Error(
+        'The form cannot be submitted without a site language');
+    }
+
+    // Do some form validation.
+    if (!this.state.name) {
+      errorHandler.addMessage(
+        i18n.gettext('Collection name cannot be blank')
+      );
+      return;
+    }
+
+    dispatch(updateCollection({
+      collectionSlug: collection.slug,
+      description: { [siteLang]: this.state.description },
+      errorHandlerId: errorHandler.id,
+      formOverlayId: COLLECTION_OVERLAY,
+      name: { [siteLang]: this.state.name },
+      user: siteUserId,
+    }));
+  }
+
+  onTextInput = (
+    event: ElementEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    event.preventDefault();
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  propsToState(props: Props) {
+    return {
+      description: props.collection && props.collection.description,
+      name: props.collection && props.collection.name,
+    };
+  }
+
+  render() {
+    const { errorHandler, i18n } = this.props;
+
+    return (
+      <FormOverlay
+        className="CollectionManager"
+        id={COLLECTION_OVERLAY}
+        onCancel={this.onCancel}
+        onSubmit={this.onSubmit}
+        submitText={i18n.gettext('Save collection')}
+        submittingText={i18n.gettext('Saving collection')}
+        title={i18n.gettext('Edit collection')}
+      >
+        {errorHandler.renderErrorIfPresent()}
+        <label htmlFor="collectionName">
+          {i18n.gettext('Collection name')}
+        </label>
+        <input
+          onChange={this.onTextInput}
+          id="collectionName"
+          name="name"
+          type="text"
+          value={this.state.name}
+        />
+        <label htmlFor="collectionDescription">
+          {i18n.gettext('Description')}
+        </label>
+        <textarea
+          defaultValue={this.state.description}
+          id="collectionDescription"
+          name="description"
+          onChange={this.onTextInput}
+        />
+      </FormOverlay>
+    );
+  }
+}
+
+export const extractId = (ownProps: Props) => {
+  const { collection } = ownProps;
+
+  return `collection-${collection ? collection.slug : ''}`;
+};
+
+export const mapStateToProps = (
+  state: {| api: ApiStateType, users: UsersStateType |}
+) => {
+  return {
+    siteLang: state.api.lang,
+    siteUserId: state.users.currentUserID,
```

Comment:
This is just a nitpick but I find `currentUserID` a better variable name to reason about something related to the current user rather than the site; it belongs to the currently signed-in user, not the site... I would rather us use `currentUser` and `currentUserId` everywhere but that's just me.

Reply:
Since there is already the concept of a *collection user* (she who owns the collection), I think it's helpful to make a distinction of *site user* (she who is using the site). Calling it *current* still seems ambiguous because it could be the current collection owner.